### PR TITLE
[SYCL][Doc] Get started guide clean-up

### PR DESCRIPTION
**Pull request by bader**
- Removed instructions to make a DPCPP_HOME/build directory
- Fix issues reported by Markdown linter

Signed-off-by: Alexey Bader <alexey.bader@intel.com>

### CODE REVIEW

File: sycl/doc/GetStartedGuide.md [hunk 1]
```diff
@@ -21,40 +21,40 @@ and a wide range of compute accelerators such as GPU and FPGA.
...
+    [Download](https://visualstudio.microsoft.com/downloads/)
 
 ### Create DPC++ workspace
 
 Throughout this document `DPCPP_HOME` denotes the path to the local directory
 created as DPC++ workspace. It might be useful to create an environment variable
 with the same name.
 
-**Linux**
+**Linux**:
```
Comment (vladimirlaz) [inline sycl/doc/GetStartedGuide.md:41]: do we need semicolomn?

Comment (bader) [reply sycl/doc/GetStartedGuide.md:41]: Yep. Otherwise linter recommends replacing it with heading.

File: sycl/doc/GetStartedGuide.md [hunk 2]
```diff
@@ -493,8 +508,8 @@ int main() {
```
Comment (pvchupin) [inline sycl/doc/GetStartedGuide.md:508]: Please update Table of Content with "Code the program for a specific GPU"

**Comment by pvchupin** [comment]
My suggestions could be done as part of this cleanup or the next cleanup. @vladimirlaz was going to address some of those as far as I remember.

File: sycl/doc/GetStartedGuide.md [hunk 3]
```diff
@@ -176,7 +173,6 @@ can be downloaded from the following web pages:
 * Windows: [Intel&reg; Download
```
[the remark this answers is not in the document]
Comment (pvchupin) [reply sycl/doc/GetStartedGuide.md:173]: To be addressed in #1699

File: sycl/doc/GetStartedGuide.md [hunk 4]
```diff
@@ -138,6 +131,10 @@ Currently, the only combination tested is Ubuntu 18.04 with CUDA 10.2 using
 a Titan RTX GPU (SM 71), but it should work on any GPU compatible with SM 50 or
 above.
 
+### Deployment
+
+TODO: add instructions how to deploy built DPC++ toolchain.
+
 ## Use DPC++ toolchain
 
 ### Using the DPC++ toolchain on CUDA platforms
```
[the remark this answers is not in the document]
Comment (bader) [reply sycl/doc/GetStartedGuide.md:140]: Done in https://github.com/intel/llvm/pull/3177.

File: sycl/doc/GetStartedGuide.md [hunk 5]
```diff
@@ -308,21 +316,25 @@ cmake -DIntel_SYCL_ROOT=%DPCPP_HOME%\deploy -DSYCL_IMPLEMENTATION=Intel_SYCL ...
```
[the remark this answers is not in the document]
Comment (bader) [reply sycl/doc/GetStartedGuide.md:316]: Done in https://github.com/intel/llvm/pull/3177.

File: sycl/doc/GetStartedGuide.md (hunk 2 above)
Comment (bader) [reply sycl/doc/GetStartedGuide.md:508]: Done in https://github.com/intel/llvm/pull/3177.

**Pull request merged**